### DISCUSSION TITLE
How do you format date

### DISCUSSION THREAD

**Discussion by jpeter06** [mockoon/mockoon#370] [Help] [answered]
Hello all, 
Just playing with mockoon, really impresed :-)

One question, how do you format date output:
I know how to do it with "now": {{now 'T'}}
but I don´t know how to do it with  "faker": {{faker 'date.recent'  ???}}

thanks for all

**Comment by 255kb** [comment, marked as answer]
Hi, we are happy that you enjoy using Mockoon 😄 
Unfortunately Faker.js date helpers do not support formatting. 
However, before switching to Faker.js we were using DummyJSON and kept some of its helpers for backward compatibility. 
What you could use is the `date` helper: 
`{{date '2020-11-20' '2020-11-25' 'dd-MM-yyyy'}}`
It requires you to set 3 params in the following order `'from' 'to' 'format'`.

**Reply by jpeter06** [reply]
Thanks Guillaume :-).
Yes, at the end I am using {{date '2020-11-17' '2020-11-17' 'T'}}  and works perfectly.
Anyway, another question,
any posiblility of obtain secuentials dates, something like:
'2020-11-17 00:00:00'
'2020-11-17 01:00:00'
'2020-11-17 02:00:00'
'2020-11-17 03:00:00'
'2020-11-17 04:00:00'

thanks for your quick response

**Reply by 255kb** [reply]
There is no way to "synchronize" multiple calls to the helpers. But maybe there is a way. If you use the `repeat` helper, you have access to `{{@index}}` inside the repeat, which give you the current index. Maybe you can use this combined to multiple `{{date ...}}` helpers (because you cannot concatenate or use `@index` in the `format` param I think). But it's kind of hacky 😄

**Reply by jpeter06** [reply]
Yes, I was trying to do something like math operations :
{{math {{now 'T'}} + @index * 3600}} or something like that
Should be posible to introduce Javascript directly?

**Reply by 255kb** [reply]
We don't have a math operator and we are constrained by Handlebars possibilities. As Handlebars idea is to be logicless we don't have this kind of options of adding, concatenating, etc. Also, JS wouldn't work without developing new helpers that does some sort of eval().

**Comment by mj9** [comment]
This one works to get a recent date, as long as you don't care about time of day:
`{{subtract (now 'T') (multiply 86400000 2)}}`
The 2 here is number of days to go back. You could also multiply with a random number
